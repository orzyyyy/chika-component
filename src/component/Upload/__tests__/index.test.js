import React from 'react';
import { mount } from 'enzyme';
let Upload;
switch (process.env.LIB_DIR) {
  case 'lib':
    Upload = require('../../../../lib/component/Upload').default;
    break;
  case 'dist':
    Upload = require('../../../../dist/lib/Upload').default;
    break;
  default:
    Upload = require('..').default;
    break;
}

const fileList = [
  { id: 1, url: 'test1', error: false },
  { id: 2, url: 'test2', error: false },
];

describe('Upload', () => {
  it('render correctly', () => {
    const wrapper = mount(<Upload fileList={fileList} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick should work correctly', () => {
    const onClick = jest.fn();
    const wrapper = mount(<Upload fileList={fileList} onClick={onClick} />);
    wrapper
      .find('.upload-item')
      .at(0)
      .simulate('click');
    expect(onClick).toBeCalledWith({ id: 1, url: 'test1', error: false });
  });

  it('status error should work correctly', () => {
    fileList[0].error = true;
    const wrapper = mount(<Upload fileList={fileList} />);
    expect(wrapper.find('.upload-item-error').length).toBe(1);
  });

  it('fileList should work correctly', () => {
    const wrapper = mount(<Upload fileList={fileList} />);
    wrapper.setProps({ fileList: [fileList[0]] });
    expect(wrapper.find('.upload-item').length).toBe(2);
  });
});
