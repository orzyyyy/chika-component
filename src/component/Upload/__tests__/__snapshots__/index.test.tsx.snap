// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload render correctly 1`] = `
<div
  className="upload"
>
  <UploadView
    fileList={
      Array [
        Object {
          "error": false,
          "id": 1,
          "url": "test1",
        },
        Object {
          "error": false,
          "id": 2,
          "url": "test2",
        },
      ]
    }
  />
</div>
`;
